ipaserver/install/server/upgrade.py

```diff
@@ -878,77 +883,50 @@ def certificate_renewal_update(ca, ds, http):
-    dirsrv_dir = dsinstance.config_dirname(serverid)
 
     # bump version when requests is changed
     version = 6
-    requests = (
-        (
-            paths.PKI_TOMCAT_ALIAS_DIR,
-            'auditSigningCert cert-pki-ca',
-            'dogtag-ipa-ca-renew-agent',
-            template % 'stop_pkicad',
-            '%s "auditSigningCert cert-pki-ca"' % (template % 'renew_ca_cert'),
-            None,
-        ),
-        (
-            paths.PKI_TOMCAT_ALIAS_DIR,
-            'ocspSigningCert cert-pki-ca',
-            'dogtag-ipa-ca-renew-agent',
-            template % 'stop_pkicad',
-            '%s "ocspSigningCert cert-pki-ca"' % (template % 'renew_ca_cert'),
-            None,
-        ),
-        (
-            paths.PKI_TOMCAT_ALIAS_DIR,
-            'subsystemCert cert-pki-ca',
-            'dogtag-ipa-ca-renew-agent',
-            template % 'stop_pkicad',
-            '%s "subsystemCert cert-pki-ca"' % (template % 'renew_ca_cert'),
-            None,
-        ),
-        (
-            paths.PKI_TOMCAT_ALIAS_DIR,
-            'caSigningCert cert-pki-ca',
-            'dogtag-ipa-ca-renew-agent',
-            template % 'stop_pkicad',
-            '%s "caSigningCert cert-pki-ca"' % (template % 'renew_ca_cert'),
-            'ipaCACertRenewal',
-        ),
-        (
-            paths.IPA_RADB_DIR,
-            'ipaCert',
-            'dogtag-ipa-ca-renew-agent',
-            template % 'renew_ra_cert_pre',
-            template % 'renew_ra_cert',
-            None,
-        ),
-        (
-            paths.PKI_TOMCAT_ALIAS_DIR,
-            'Server-Cert cert-pki-ca',
-            'dogtag-ipa-ca-renew-agent',
-            template % 'stop_pkicad',
-            '%s "Server-Cert cert-pki-ca"' % (template % 'renew_ca_cert'),
-            None,
-        ),
-        (
-            paths.HTTPD_ALIAS_DIR,
-            'Server-Cert',
-            'IPA',
-            None,
-            template % 'restart_httpd',
-            None,
-        ),
-        (
-            dirsrv_dir,
-            'Server-Cert',
-            'IPA',
-            None,
-            '%s %s' % (template % 'restart_dirsrv', serverid),
-            None,
-        ),
-
-    )
+    requests = []
+    for certnick in ('auditSigningCert cert-pki-ca',
+                     'ocspSigningCert cert-pki-ca',
+                     'subsystemCert cert-pki-ca',
+                     'caSigningCert cert-pki-ca',
+                     'Server-Cert cert-pki-ca',
+                     ):
+        requests.append(
+            {
+                'cert-database': paths.PKI_TOMCAT_ALIAS_DIR,
+                'cert-nickname': certnick,
+                'ca': 'dogtag-ipa-ca-renew-agent',
+                'cert-presave-command': template % 'stop_pkicad',
+                'cert-postsave-command':
+                    (template % 'renew_ca_cert "{}"').format(certnick),
+            }
+        )
+    # set profile for caSigningCert
+    requests[3]['template-profile'] = 'ipaCACertRenewal'
```

Comment:
I have to say I didn't like having it on two places very much as well. Could have done it similar to what's in `CAInstance.tracking_reqs` but did not like that very much either. Will keep it the old way with a minor improvement of having the requests there straight away.